edb/schema/extensions.py

```diff
@@ -548,9 +565,20 @@ def filt(schema: s_schema.Schema, obj: so.Object) -> bool:
             # scalar types. Cleaning casts earlier may cause issues with
             # functions that use casts in their signatures as part of the
             # default expression.
-            if isinstance(subast, qlast.DropScalarType) and casts_cleanup:
-                self.update(casts_cleanup)
-                casts_cleanup.clear()
+            #
+            # For the same reasons we also want to clean up index matches
+            # before we're cleaning indexes or scalar types.
```

Comment:
Can we just always drop the index matches first? We don't need to have them since they can't be used in functions like casts

Reply:
You're right, I've moved the index matches to be dropped first.